File: LarkCardKit/Builders/ImageBuilder.cs
using LarkCardKit.Enums;
using LarkCardKit.Models.Elements;

namespace LarkCardKit.Builders;

public class ImageBuilder
{
    private readonly Image _image;
    
    public ImageBuilder(string imgKey)
    {
        _image = new Image { ImgKey = imgKey };
    }
    
    public ImageBuilder Title(string title)
    {
        _image.Title = new PlainText { Content = title };
        return this;
    }
    
    public ImageBuilder ScaleType(string scaleType)
    {
        _image.ScaleType = scaleType;
        return this;
    }
    
    public ImageBuilder Size(ImageSize size)
    {
        _image.Size = size switch
        {
            ImageSize.CropCenter => "crop_center",
            ImageSize.CropTop => "crop_top",
            ImageSize.CropBottom => "crop_bottom",
            ImageSize.Origin => "origin",
            _ => "crop_center"
        };
        return this;
    }
    
    public ImageBuilder Size(string size)
    {
        _image.Size = size;
        return this;
    }
    
    public ImageBuilder CornerRadius(string cornerRadius)
    {
        _image.CornerRadius = cornerRadius;
        return this;
    }
    
    public ImageBuilder Transparent(bool transparent = true)
    {
        _image.Transparent = transparent;
        return this;
    }
    
    public ImageBuilder Preview(bool preview = true)
    {
        _image.Preview = preview;
        return this;
    }
    
    public ImageBuilder Alt(string alt)
    {
        _image.Alt = new PlainText { Content = alt };
        return this;
    }
    
    public ImageBuilder Width(string width)
    {
        _image.Width = width;
        return this;
    }
    
    public ImageBuilder ElementId(string id)
    {
        _image.ElementId = id;
        return this;
    }
    
    public ImageBuilder Margin(string margin)
    {
        _image.Margin = margin;
        return this;
    }
    
    public Image Build() => _image;
}

File: TypeScript/LarkCardKit/src/builders/ImageBuilder.ts
import type { Image } from '../models/elements';
import { ImageSize } from '../enums';

export class ImageBuilder {
  private _image: Image;

  constructor(imgKey: string) {
    this._image = { tag: 'img', img_key: imgKey };
  }

  alt(alt: string): this {
    this._image.alt = { tag: 'plain_text', content: alt };
    return this;
  }

  size(size: ImageSize | string): this {
    this._image.size = typeof size === 'string' ? size : size;
    return this;
  }

  mode(mode: string): this {
    this._image.mode = mode;
    return this;
  }

  width(width: string): this {
    this._image.width = width;
    return this;
  }

  height(height: string): this {
    this._image.height = height;
    return this;
  }

  preview(preview: boolean = true): this {
    this._image.preview = preview;
    return this;
  }

  hoverTips(tips: string): this {
    this._image.hover_tips = { tag: 'plain_text', content: tips };
    return this;
  }

  clickUrl(url: string): this {
    this._image.click_url = url;
    return this;
  }

  elementId(id: string): this {
    this._image.element_id = id;
    return this;
  }

  margin(margin: string): this {
    this._image.margin = margin;
    return this;
  }

  build(): Image {
    return this._image;
  }
}
